### PR TITLE
Bluespace Harvester Resprite

### PR DESCRIPTION
**Pull request by kugamo**
## What Does This PR Do
Resprites the Bluespace Harvester.

Also shifts its center to its center (items would spawn offcenter otherwise)...
but this also means it should be wired to there instead of its upper center tile now.

Also also slightly changes the redspace portals event, so that the portals spawn sequentially over at most 15 seconds. 
The BSH will glow red and be inoperable until they all are destroyed.

## Why It's Good For The Game
Sprite Improvement.

## Images of changes

https://github.com/ParadiseSS13/Paradise/assets/16618648/e11d5d28-36c6-4c88-a4d6-4869038747fd

dreamseeker_bg89MN9Kkx

Aseprite_HwGksibJs9

## Testing
I vareditted a SMES to get the different power levels required to check all the sprites are working at the appropriate levels.

Also found out BSH really doesnt like its wire that feeds it to be cut, but that seems like a separate issue.

## Changelog
:cl:
tweak: Bluespace Harvester's center is now in the center of its sprite.
imageadd: New sprites for the Bluespace Harvester.
/:cl:

### CODE REVIEW

**Comment by kugamo** [comment]
> Please tell me it goes red once it starts spawning the netherworld portals

Its funny you mention it, I just added this today!

https://github.com/ParadiseSS13/Paradise/assets/16618648/a3a6983d-fbbd-4f76-ad56-4282d911ab5d

**Comment by Christasmurf** [comment]
> > Please tell me it goes red once it starts spawning the netherworld portals
> 
> Its funny you mention it, I just added this today!
> 
>  dreamseeker_SME78cgvBR.mp4

YOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO

File: code/modules/station_goals/bluespace_tap.dm [hunk 1]
```diff
@@ -428,9 +497,13 @@
 	spawn_time = 30 SECONDS
 	max_mobs = 5		//Dont' want them overrunning the station
 	max_integrity = 250
+	var/obj/machinery/power/bluespace_tap/linked_source_object //this is the BSH that caused it
 
 /obj/structure/spawner/nether/bluespace_tap/deconstruct(disassembled)
 	new /obj/item/stack/ore/bluespace_crystal(loc)	//have a reward
+	if(linked_source_object)
+		linked_source_object.active_nether_portals -= src
+		linked_source_object.update_icon(UPDATE_ICON)
```
Comment (Sirryan2002) [inline code/modules/station_goals/bluespace_tap.dm:506]: put this in `Destroy()` instead, if portals ever get deleted via a non `deconstruct()` method then they will not be properly garbage collected

File: code/modules/station_goals/bluespace_tap.dm [hunk 2]
```diff
@@ -428,9 +497,13 @@
 	spawn_time = 30 SECONDS
 	max_mobs = 5		//Dont' want them overrunning the station
 	max_integrity = 250
+	var/obj/machinery/power/bluespace_tap/linked_source_object //this is the BSH that caused it
```
Comment (Sirryan2002) [inline code/modules/station_goals/bluespace_tap.dm:500]:
```suggestion
	/// the BSH that spawned this portal
	var/obj/machinery/power/bluespace_tap/linked_source_object 
```

**Review by Sirryan2002 — APPROVED** [review]

File: code/modules/station_goals/bluespace_tap.dm [hunk 3]
```diff
@@ -255,6 +255,61 @@
...
+	else
+		set_light(1, 1, "#353535")
+
+	if(get_available_power() <= 0)
+		icon_state = base_icon_state
+	else
+		icon_state = "[base_icon_state][get_icon_state_number()]"
+		overlays += icon(icon, "screen")
+		if(light)
+			underlays += emissive_appearance(icon, "light_mask")
```
Comment (S34NW) [inline code/modules/station_goals/bluespace_tap.dm:278]: This should probably be broken into the new specific procs (`update_icon_state()` and `update_overlays()`) with underlay logic included in the overlays proc.

File: code/modules/station_goals/bluespace_tap.dm [hunk 4]
```diff
@@ -255,6 +255,61 @@
...
+			return 5
+
+/obj/machinery/power/bluespace_tap/power_change()
+	. = ..()
+	if(stat & (BROKEN|NOPOWER))
+		set_light(0)
+	else
+		set_light(1, 1, "#353535")
+	if(.)
+		update_icon(UPDATE_OVERLAYS)
```
Comment (S34NW) [inline code/modules/station_goals/bluespace_tap.dm:302]:
```suggestion
		update_icon(UPDATE_ICON)
```
Why `UPDATE_OVERLAYS` here? You handle all the logic in `update_icon` above

File: code/modules/station_goals/bluespace_tap.dm [hunk 5]
```diff
@@ -255,6 +255,61 @@
...
+
+
+/obj/machinery/power/bluespace_tap/connect_to_network()
+	..()
+	update_icon()
```
Comment (S34NW) [inline code/modules/station_goals/bluespace_tap.dm:307]:
```suggestion
	update_icon(UPDATE_ICON)
```

File: code/modules/station_goals/bluespace_tap.dm [hunk 6]
```diff
@@ -255,6 +255,61 @@
...
+
+/obj/machinery/power/bluespace_tap/disconnect_from_network()
+	..()
+	update_icon()
```
Comment (S34NW) [inline code/modules/station_goals/bluespace_tap.dm:311]:
```suggestion
	update_icon(UPDATE_ICON)
```

**Review by S34NW — CHANGES_REQUESTED** [review]

File: code/modules/station_goals/bluespace_tap.dm [hunk 7]
```diff
@@ -255,6 +255,75 @@
...
+		set_light(15, 5, "#ff0000")
+		return
+
+	if(stat & (BROKEN|NOPOWER))
+		set_light(0)
+	else
+		set_light(1, 1, "#353535")
+
+	if(get_available_power())
+		overlays += icon(icon, "screen")
```
Comment (Vi3trice) [inline code/modules/station_goals/bluespace_tap.dm:289]:
```suggestion
	underlays.Cut()

	if(length(active_nether_portals))
		. += "redspace"
		set_light(15, 5, "#ff0000")
		return

	if(stat & (BROKEN|NOPOWER))
		set_light(0)
	else
		set_light(1, 1, "#353535")

	if(get_available_power())
		. += "screen"
```

File: code/modules/station_goals/bluespace_tap.dm [hunk 8]
```diff
@@ -255,6 +255,75 @@
...
+			return 5
+
+/obj/machinery/power/bluespace_tap/power_change()
+	. = ..()
+	if(stat & (BROKEN|NOPOWER))
+		set_light(0)
+	else
+		set_light(1, 1, "#353535")
+	if(.)
+		update_icon(UPDATE_ICON)
```
Comment (Vi3trice) [inline code/modules/station_goals/bluespace_tap.dm:316]: `update_icon()` by default uses `UPDATE_ICON`, so you don't need to put that everywhere. What you want to use is either `UPDATE_ICON_STATE` or `UPDATE_OVERLAYS` if you know of any situation where only one of the two is being affected.

This prevents unnecessary updating of the icon state or overlays unless you're updating both on purpose.

Comment (kugamo) [reply code/modules/station_goals/bluespace_tap.dm:316]: @S34NW

**Review by S34NW — APPROVED** [review]

**Comment by ThePissGuy** [comment]
stolen

**Pull request merged**